_posts/2023-10-06-k8s-comparison-azure-aws.md

```diff
@@ -0,0 +1,264 @@
+authors: [pieter_vincken]
+title: 'Are all managed Kubernetes clusters created equally?'
+image: /img/2023-10-06-k8s-comparison-azure-aws/header.png
+tags: [cloud, automation, cicd, terraform, iac, azure, aws]
+category: Cloud
+comments: false
+---
+
+> Like stars in the night sky, managed Kubernetes services vary in brilliance; not all shine equally, but the right one can illuminate your journey to cloud-native excellence. - ChatGPT 2023
+
+- [Introduction](#introduction)
+- [The use case](#the-use-case)
+- [Comparison](#comparison)
+- [Architectural components for Kubernetes setup](#architectural-components-for-kubernetes-setup)
+- [How managed is the managed service?](#how-managed-is-the-managed-service)
+- [Out of the box support](#out-of-the-box-support)
+- [Conclusion](#conclusion)
+
+# Introduction
+
+Kubernetes has been the de-facto standard for cloud-native container orchestration for some years now.
+Organizations that rely on a microservices architecture, often rely on Kubernetes to be their platform to deploy to. 
+One of the biggest selling points, especially in recent years, is that Kubernetes can provide a very similar user experience for developers across different on-premise and cloud environments.
+It can provide a common abstraction from the underlying infrastructure through its API. 
+This also allows solutions built on top of Kubernetes to be easily portable across different environments.
+Or at least, that's the promise that is being made by Kubernetes platform teams across the industry.
+
+Running Kubernetes yourself can be very hard to do properly.
+Managing all the moving parts, especially when the workloads are changing frequently can be challenging, to say the least.
+Mix in the usage of more advanced features like a service mesh, custom operators or multi-cloud cluster and it can become a real challenge to maintain.
+All hyper-scale (and more and more smaller scale) cloud providers have managed Kubernetes offerings for customers who want to leverage the power of Kubernetes, but don't want the hassle of managing all the different pieces of the solution. 
+Google has its Google Kubernetes Engine, GKE for short, Azure has its Azure Kubernetes Service, AKS, and lastly, Amazon has its Elastic Kubernetes Service, EKS for short. 
+Although all of these services deliver a Kubernetes cluster, they aren't all created equally as will become clear further in this blog post.
+
+For most deployments of Kubernetes, Kubernetes itself is only part of the software stack of a software project. 
+A plethora of extensions, add-ons and so-called operators are used to extend the default functionality and integrate external services into Kubernetes for easier management of those resources. 
+Some common examples are the `external-dns` operator, `certmanager` and the `cluster auto scaler` projects. 
+These all add additional functionality, DNS management, Certificate Management and auto-scaling capabilities respectively, to the cluster. 
```

Comment:
Horizontaal pods scalen wordt gedaan met metrics-server, mss hier ook vermelden?

Reply:
HPA is niet echt een operator maar meer een "native" feature toch? Ik wil hier maar aantonen dat operators de functionaliteit van een cluster kunnen extenden.

Reply:
> HPA is niet echt een operator maar meer een "native" feature toch? Ik wil hier maar aantonen dat operators de functionaliteit van een cluster kunnen extenden.

Daar volg ik je in, cava!